From 239a8ab5a2cf9d71029df5ec247c46ac5b5e8cff Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:08 -0400
Subject: [powerpc] Tell about irq stack coverage

Message-id: <1410545655-205645-80-git-send-email-dzickus@redhat.com>
Patchwork-id: 94091
O-Subject: [RHEL7 PATCH 079/626] powerpc: Tell about irq stack coverage
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 62d26c8200a8382e1c67419ca3aff78d37898cc5
Author: Frederic Weisbecker <fweisbec@gmail.com>
Date:   Tue Sep 24 17:18:36 2013 +0200

    powerpc: Tell about irq stack coverage

    Now that powerpc runs irq_exit() under the irq stack,
    let the softirq core know about that so that we spare
    the needless stack switch on irq exit's softirq processing.

    Acked-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
    Acked-by: Linus Torvalds <torvalds@linux-foundation.org>
    Signed-off-by: Frederic Weisbecker <fweisbec@gmail.com>
    Cc: Benjamin Herrenschmidt <benh@kernel.crashing.org>
    Cc: Paul Mackerras <paulus@au1.ibm.com>
    Cc: Ingo Molnar <mingo@kernel.org>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: H. Peter Anvin <hpa@zytor.com>
    Cc: Linus Torvalds <torvalds@linux-foundation.org>
    Cc: Paul Mackerras <paulus@au1.ibm.com>
    Cc: James Hogan <james.hogan@imgtec.com>
    Cc: James E.J. Bottomley <jejb@parisc-linux.org>
    Cc: Helge Deller <deller@gmx.de>
    Cc: Martin Schwidefsky <schwidefsky@de.ibm.com>
    Cc: Heiko Carstens <heiko.carstens@de.ibm.com>
    Cc: David S. Miller <davem@davemloft.net>
    Cc: Andrew Morton <akpm@linux-foundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/Kconfig b/arch/powerpc/Kconfig
index e9f9fe1..44d84a2 100644
--- a/arch/powerpc/Kconfig
+++ b/arch/powerpc/Kconfig
@@ -138,6 +138,7 @@ config PPC
  select ARCH_USE_BUILTIN_BSWAP
  select OLD_SIGSUSPEND
  select OLD_SIGACTION if PPC32
+ select HAVE_IRQ_EXIT_ON_IRQ_STACK
 
 config GENERIC_CSUM
  def_bool CPU_LITTLE_ENDIAN
-- 
1.7.1